pulp_2to3_migration/app/plugin/rpm/pulp_2to3_models.py

```diff
@@ -535,7 +546,28 @@ async def create_pulp3_content(self):
         """
         Create a Pulp 3 Distribution content for saving later in a bulk operation.
         """
-        pass
+        namespaces = [".treeinfo", "treeinfo"]
+        for namespace in namespaces:
+            treeinfo = PulpTreeInfo()
+            try:
+                treeinfo.load(f=os.path.join(self.pulp2content.pulp2_storage_path, '.treeinfo'))
+            except FileNotFoundError:
+                continue
+            self.filename = namespace
+            treeinfo_parsed = treeinfo.parsed_sections()
+            treeinfo_serialized = TreeinfoData(treeinfo_parsed).to_dict(filename=namespace)
+            # The repository must already be migrated in order to migrate the DistributionTree
+            pulp2_r_c_unit = RepositoryContentUnit.objects.get(unit_id=self.pulp2content.pulp2_id)
+            pulp2_repo_id = pulp2_r_c_unit.repo_id
+            pulp2_repo = Pulp2Repository.objects.get(pulp2_repo_id=pulp2_repo_id)
```

Comment:
Since pulp2_repo_id is not unique in the span of multiple migration re-runs (a repo in pulp2 can be removed and a new, completely different one, can be created with the same repo_id), an additionla condition is needed: `not_in_plan=False`.
It means that you are looking for a repository which is expected to be present/migrated as a part of the current migration plan.

Reply:
Cool. I'll add that condition.